components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementServiceImpl.java

```diff
@@ -0,0 +1,666 @@
+
+        return roleDAO.getRoleIdListOfUser(userId, tenantDomain);
+    }
+
+    @Override
+    public List<String> getRoleIdListOfGroups(List<String> groupIds, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return roleDAO.getRoleIdListOfGroups(groupIds, tenantDomain);
+    }
+
+    @Override
+    public List<String> getRoleIdListOfIdpGroups(List<String> groupIds, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return roleDAO.getRoleIdListOfIdpGroups(groupIds, tenantDomain);
+    }
+
+    private String getUser(String tenantDomain) {
+
+        String user = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isNotBlank(user)) {
+            user = UserCoreUtil.addTenantDomainToEntry(user, tenantDomain);
+        } else {
+            user = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        return user;
+    }
+
+    /**
+     * Get organization ID by tenantDomain.
+     *
+     * @param tenantDomain tenantDomain.
+     * @throws IdentityRoleManagementException Error occurred while retrieving organization id.
+     */
+    private String getOrganizationIdByTenantDomain(String tenantDomain) throws IdentityRoleManagementException {
+
+        try {
+            return RoleManagementServiceComponentHolder.getInstance().getOrganizationManager()
+                    .resolveOrganizationId(tenantDomain);
+
+        } catch (OrganizationManagementException e) {
+            String errorMessage = "Error while retrieving the organization id for the given tenantDomain: "
+                    + tenantDomain;
+            throw new IdentityRoleManagementServerException(UNEXPECTED_SERVER_ERROR.getCode(), errorMessage, e);
+        }
+    }
+
+    /**
+     * Validate application role audience.
+     *
+     * @param applicationId Application Id.
+     * @throws IdentityRoleManagementException Error occurred while validating application role audience.
+     */
+    private void validateApplicationRoleAudience(String applicationId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        try {
+            ServiceProvider app = RoleManagementServiceComponentHolder.getInstance().getApplicationManagementService()
+                    .getApplicationByResourceId(applicationId, tenantDomain);
+            if (app == null) {
+                throw new IdentityRoleManagementClientException(INVALID_AUDIENCE.getCode(),
+                        "Invalid audience. No application found with application id: " + applicationId);
+            }
+            boolean valid = false;
+            for (ServiceProviderProperty property : app.getSpProperties()) {
+                if ("ROLE_AUDIENCE_TYPE".equals(property.getName())) {
+                    if (APPLICATION.equalsIgnoreCase(property.getValue())) {
+                        valid = true;
+                    }
```

Comment:
Let's use https://github.com/wso2/carbon-identity-framework/pull/5001/files#diff-9c35646b8cf81e67a1751aa918724dc41f9029ab886488277ba4624adb8ba32dR526 this